pkg/clustermesh/ipsetsyncer.go

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package clustermesh
+
+import (
+	"context"
+	"runtime/pprof"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/cilium/cilium/pkg/datapath/iptables/ipset"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/hive/job"
+)
+
+func ipsetSyncer(
+	logger logrus.FieldLogger,
+	lc cell.Lifecycle,
+	jobRegistry job.Registry,
+	scope cell.Scope,
+	cm *ClusterMesh,
+	ipsetMgr ipset.Manager,
+) {
+	if cm == nil {
+		return
+	}
+
+	initializer := ipsetMgr.NewInitializer()
+
+	jg := jobRegistry.NewGroup(
+		scope,
+		job.WithLogger(logger),
+		job.WithPprofLabels(pprof.Labels("cell", "clustermesh-ipset-syncer")),
+	)
+	jg.Add(job.OneShot("clustermesh-ipset-syncer", func(ctx context.Context, _ cell.HealthReporter) error {
+		// wait for initial nodes listing from all remote clusters
+		// before allowing stale ipset entries deletion
+		cm.NodesSynced(ctx)
```

Comment:
Nit. It is likely better to check the possible error, and return in that case. The only possible error is if the context gets closed, but in that case we don't want to accidentally trigger the pruning step due to race conditions (although theoretically the corresponding context should be canceled as well).

Reply:
I'll specifically check for the `context.Canceled` in order to avoid re-triggering the job :+1: